packit/api.py

```diff
@@ -173,6 +173,11 @@ def sync_release(
         """
         assert_existence(self.up.local_project)
         assert_existence(self.dg.local_project)
+        if self.dg.is_dirty():
```

Comment:
I recall you talked about this yesterday: what's the precise problem we are solving here?

Reply:
This is just a first step for making sure the dist-git branch is clean before syncing. This change is more related to the usage via CLI.

In packit-service I am making PR to reset the changes (e.g. from a failed sync), that prevented other branches to be synced: https://sentry.io/organizations/red-hat-0p/issues/1444893591

Reply:
danke